### DISCUSSION TITLE
Shortest way to get all "Playing" Groups

### DISCUSSION THREAD

**Discussion by jnbht** [hklages/node-red-contrib-sonos-plus#141] [Q&A] [answered]
Hey, i want to group a player with a wall switch.
To do this, i need to know, which groups/players are actually playing something.

My way right now is to get the household an go through this info with for-loops until i know which player is doing its money worth.

Is there a easier way? 

F.e. Pushing and holding the Play-button on a Sonos speaker groups the player with the next playing group. Push and holding again, with the next and so on.
Is there a special topic for this?

**Comment by hklages** [comment, marked as answer]
No - there is no special solution for that. Your for-loop or split/join is a good approach. You have to check all players with index 0 in group - they are the coordinators. 

Another option would be to use the sister package node-red-contrib-sonos-events and subscribe to all coordinators with AVTransport, playbackstate, store for each one the playback state in a global variable. Then you only have to check this variable. Thats maybe the fastest method. Group changes you may monitor with Zongroup event.

**Reply by jnbht** [reply]
Ok, thank you. Then for-loops it shall be ;)
